core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -527,13 +527,15 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       }
     }
 
-    // New consumer always hasNext
-    override def hasData = true
-
-    override def receive() : BaseConsumerRecord = {
-      while (recordIter == null || !recordIter.hasNext)
+    override def hasData : Boolean = {
+      if (recordIter == null || !recordIter.hasNext) {
         recordIter = consumer.poll(1000).iterator
+      }
 
+      recordIter.hasNext
+    }
+
+    override def receive() : BaseConsumerRecord = {
```

Comment:
I took a closer look at mirror maker. Reusing ```NewShinyConsumer``` and ```OldConsumer``` is much harder than it sounds. Here are the findings

- A consumerRebalanceListener is needed for both new consumer and old consumer for no data loss purpose. ConsumerRebalanceListener itself requires reference to consumer to commit offset hence we need to create a new method in ```BaseConsumer``` called subscribe or something that takes ConsumerRebalanceListener as parameter and set it to consumer.  But new and old consumer expect different interfaces of ConsumerRebalanceListener that makes it hard to define the subscribe method in ```BaseConsumer``` plus old consumer expects TopicFilter to subscribe while new consumer expects topic or whitelist (string).
- Offset commit behavior is different between new and old consumers in mirror maker. In new consumer we need to pass in a Java offset map to indicate which offset is ready to commit while old consumer doesn't that as the consumer iterator is able to set the latest consumed offset. 

Given the fact that we will retire old consumer eventually and ConsoleConsumer and mirror maker are the only two use cases that needs combination of new/old consumers. I'd like to keep this fix as minimal as possible. I will update the fix to reuse ```NewShinyConsumer``` receive() logic to make it consistent with old consumer.

Thoughts? @ijuma @hachikuji @becketqin